src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -113,28 +96,31 @@ public void storeMessage(Message message) {
     // All the messages between the two users
     Filter directMessages = CompositeFilterOperator.or(loggedInUserMessages, otherUserMessages);
 
+    // Sort in the ascending order so that most recent messages appear last
     Query query =
         new Query("Message")
             .setFilter(directMessages)
             .addSort("timestamp", SortDirection.ASCENDING);
     PreparedQuery results = datastore.prepare(query);
 
     for (Entity entity : results.asIterable()) {
+
+      String idString = null;
       try {
```

Comment:
When can `entity` or `entity.getKey()` be null? To my understanding, this is a situation that should not be possible (every entity that has been successfully retrieved from the database has a non-null key).

I would recommend simplifying the entire `try`...`catch` block you have here to:

     String idString = entity.getKey().getName();